chromadb/experimental/examples/density.ipynb

```diff
@@ -0,0 +1,471 @@
+    "require significant tuning and experimentation for each new dataset, and the right threshold is likely to vary \n",
+    "with the data. \n",
+    "\n",
+    "## Just ask the model ?\n",
+    "\n",
+    "One approach to this problem is to just ask the model whether information in its context is relevant to the query.\n",
+    "This might work well in the general case, but comes with some drawbacks. \n",
+    "\n",
+    "LM calls are relatively expensive compared to querying Chroma. Additionally, in many use-cases, the model may not be \n",
+    "able to assess the relevancy of a document to a given query, especially if there are few or no examples of these types \n",
+    "of documents in the training corpus. \n",
+    "\n",
+    "## Algorithmic relevance \n",
+    "\n",
+    "Density is an algorithmic approach to the problem of query relevance, which automatically adapts to the dataset and\n",
+    "embedding model. In brief, Chroma computes a distance percentile for each query result and returns it as an auxilliary \n",
+    "output along with the query result itself. Here's how it works: \n",
+    "\n",
+    "- For each embedding in the dataset, we compute the distances to N nearest neighbors. \n",
+    "- We construct a cumulative density over the returned distances, and store it as a lookup table. \n",
+    "\n",
+    "When a query arrives:\n",
+    "\n",
+    "- We get the distance to the query for each result.\n",
+    "- We use the cumulative density lookup table to determine the percentile each distance lies in relative to the dataset \n",
+    "as a whole.\n",
+    "- We return this percentile $p$ as an auxilliary output. \n",
+    "\n",
+    "The percentile $p$ can be thought of as a proxy for the probability that a given neighbor is relevant to the query. We\n",
+    "can then use this proxy value to decide whether a result is relevant before giving it to the LM.\n",
+    "\n",
+    "See [this post]() for a more in-depth write up. On to the examples! "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import urllib.request\n",
+    "import zipfile\n",
+    "\n",
+    "import chromadb.experimental\n",
+    "from chromadb.config import Settings"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Dataset\n",
+    "\n",
+    "As an example dataset we use an embedding of the bible. \n",
+    "\n",
+    "We embed the bible is by verse, using the `multi-qa-mpnet-base-dot-v1` embedding model \n",
+    "from [Sentence Transformers](https://huggingface.co/sentence-transformers/multi-qa-mpnet-base-dot-v1). This model is \n",
+    "trained such that short questions and their accompanying answers lie close together in emebedding space, \n",
+    "under the dot product. \n",
```

Comment:
"under the dot product" is too low level - what does this mean practically for users, teach them why should they pick this embedding over the default